src/base/NekRSProblem.C

```diff
@@ -828,4 +919,45 @@ NekRSProblem::addExternalVariables()
   }
 }
 
+void
+NekRSProblem::calculateMeshVelocity(int e, const field::NekWriteEnum & field)
+{
+  int len = _nek_mesh->order() + 2;
+  double *tmp_meshvel = (double *)calloc(len*len, sizeof(double));
+  double dt = _timestepper->getCurrentDT();
+
+  switch (field)
+  {
+    case field::mesh_velocity_x:
+      for (int i=0; i <len * len; i++)
+      {
+        tmp_meshvel[i] = ( _displacement_x[i]-_nek_mesh->prev_disp_x().at((e*len*len) + i))
+                           /dt/_U_ref;
+      }
+      writeBoundarySolution(e, field::mesh_velocity_x,tmp_meshvel);
```

Comment:
I recently changed `writeBoundarySolution`, because I was encountering some slow performance for a big case. 

Before, `writeBoundarySolution` would first check if the MPI rank matched the elements we're trying to write. But this can be slow, if you go through all the effort to figure out `tmp_meshvel` only to not even write because the MPI rank isn't a match. So I moved the rank check outside of `writeBoundarySolution`.

To be consistent, you'll need to put 

```
        // We can only write into the nekRS scratch space if that face is "owned" by the current process
        if (nekrs::commRank() != _nek_mesh->boundaryCoupling().processor_id(e))
          continue;
```

at the start of this function and any others that are calling `writeBoundarySolution`. Same thing goes for `writeVolumeSolution`.